addons/sale_stock/models/account_invoice.py

```diff
@@ -17,6 +17,34 @@ def _get_last_step_stock_moves(self):
             rslt += invoice.mapped('refund_invoice_id.invoice_line_ids.sale_line_ids.order_id.picking_ids.move_lines').filtered(lambda x: x.state == 'done' and x.location_id.usage == 'customer')
         return rslt
 
+    def _get_invoice_snln_values(self):
+        self.ensure_one()
+        so = self.mapped('invoice_line_ids.sale_line_ids.order_id')
+        stock_move_lines = so.mapped('picking_ids.move_lines.move_line_ids')
+        last_invoice = so.invoice_ids.sorted(lambda i: i.date_invoice)[len(so.invoice_ids) - 1]
+        incoming_sml = stock_move_lines.filtered(lambda m: m.state == 'done' and m.location_id.usage == 'customer'
+                                                          and last_invoice.date_invoice <= m.date.date() <= fields.Date.today())
+        outgoing_sml = stock_move_lines.filtered(lambda m: m.state == 'done' and m.location_dest_id.usage == 'customer'
+                                                          and last_invoice.date_invoice <= m.date.date() <= fields.Date.today())
```

Comment:
you have to get the range between last invoice (if any) and self.date_invoice, not tday